Aligned sequence function for pairwise alignments and unified formatting

Ellmen opened the pull request:
This pull request addresses issue #2632 

I used the new method in the formatting function and unified `__format__` with `_format_generalized`

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

mdehoon commented:
@Ellmen  It looks OK, but I guess we could use `__getitem__` instead of `aligned_seq` such that `alignment[i]` will return the aligned sequence.
Can you also update the chapter in the tutorial about pairwise alignments?

peterjc commented:
If you want to combine this pairwise class with the MSA one day, it would be unfortunate if we had ``alignment[0]`` returning string (here), versus ``SeqRecord`` (MSA)?

peterjc commented:
Another thought would be API similarity to SearchIO, where the results are often essentially pairwise sequence alignments?

mdehoon commented:
@peterjc  The `MultipleSeqAlignment` class has some design flaws; storing the alignment as a sequence is one of them. We'll get to that after `SeqRecord`s have been unified with `Seq` records.

Ellmen commented:
I updated the method to `__getitem__` and added an entry to the docs. I also fixed the tests (they were testing for an old specification) and fixed a little formatting bug I missed. It doesn't make a huge difference to me what the type is so whatever you think is best. I can also probably help with merging PairwiseAlignment and MSA if that's relatively high priority.

mdehoon commented on Tests/test_pairwise_aligner.py:
```diff
@@ -308,7 +308,7 @@ def test_smithwaterman(self):
  AwBw
- |-| 
+ |-|.
 zA-Bz
```
This changes the interpretation of the alignment:
` |-| ` means that `AwB` was aligned to `A-B`
` |-|.` means that `AwBw` was aligned to `A-Bz` (with `w` and `z` mismatched to each other).
If you look at the path of the alignment through the trace matrix:
```
>>> alignment.path
((0, 1), (1, 2), (2, 2), (3, 3))
```
you will see that the alignment stops after the third letter. So ` |-| ` is correct here.

Same for the other alignments below I guess.

mdehoon commented:
superseded by #3565
